java/src/main/java/com/google/appengine/tools/mapreduce/impl/shardedjob/ShardedJobRunner.java

```diff
@@ -248,14 +226,19 @@ public void completeShard(final String jobId, final String taskId) {
       if (jobState.getStatus().getStatusCode() == DONE) {
         log.info("Calling completed for " + jobId);
         // TODO(user): consider trying failed if completed failed after N attempts
-        callCompleted(jobState);
+
+        //q: should this be same txn as above??
```

Comment:
I think so; I understand that all operations done here are base on a transaction, so if something is wrong in the completion/commits, that transaction should fail too...

Reply:
It looks like it should be the same...